app/frontend/components/emoji_picker.js

```diff
@@ -0,0 +1,189 @@
+    this.editor = null;
+    if (picker) {
+      picker.hidePicker();
+    }
+  }
+
+  /** 
+   * Appends emoji picker button to codeMirror 
+   * @param codeMirror - active codeMirror editor instance
+   */
+  attachEditor(codeMirror) {
+    this.editor = codeMirror;
+
+    this.style.bottom = '0.8em';
+  }
+
+  /** 
+   * Insert emoji into input or codeMirror depending on the context in the DOM
+   * To be called only by the Picker library in the "emoji" event callback
+   * @param {string} emoji - active codeMirror editor instance
+   */
+  insertEmoji(emoji) {
+    const input = this.previousElementSibling;
+    if (input && input.nodeName === 'INPUT') {
+      input.value += emoji;
+    } else if (this.editor && this.editor.getDoc()) {
+      this.editor.getDoc().replaceSelection(emoji);
+    } else {
+      console.log('EmojiPicker: Input is not detected. Can\'t add emoji', emoji);
+    }
+  }
+
+  _handleClick() {
+    if (picker === undefined) {
+      initPicker();
+    }
+    picker.togglePicker(this);
+  }
+}
+
+customElements.define('emoji-picker', EmojiTrigger);
+module.exports.EmojiTrigger = EmojiTrigger;
+
+// proof of concept; utilizing tagged templates https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates
+// FIXME: move to utility module or use the npm package
+function html(literals, ...substs) {
+  const template = document.createElement('template');
+  // based upon https://github.com/AntonioVdlC/html-template-tag/blob/main/src/index.ts
+  template.innerHTML = literals.raw.reduce((acc, lit, i) => {
+    let subst = substs[i - 1];
+    if (Array.isArray(subst)) {
+      subst = subst.join("");
+    }
+    return acc + subst + lit;
+  });
+
+  return template;
+}
+
+var picker;
+
+function initPicker() {
+  const { EmojiButton } = require('../../../node_modules/@joeattardi/emoji-button/dist/index.cjs.js');
+  const nightModeOption = app_controller.optionsMenu._nightModeOption;
+
+  picker = new EmojiButton({ // https://emoji-button.js.org/docs/api
+    emojiData: getLocalizedData(i18n.language),
+    showPreview: false,
+    showVariants: false,
+    showAnimation: false,
+    categories: ['smileys', 'people', 'animals', 'food', 'activities', 'travel', 'objects', 'symbols'],
+    i18n: i18n.t('emoji', { returnObjects: true }),
+    theme: nightModeOption && nightModeOption.isChecked ? 'dark' : 'light',
+    emojiSize: '1.3em',
+    position: 'auto',
+    zIndex: 10000,
+    styleProperties: {
+      '--background-color': '#f2f5f7',
+      '--dark-background-color': '#1e1e1e',
+    }
+  });
+
+  picker.on('emoji', ({ emoji, name, trigger }) => {
+    if (!trigger || trigger.nodeName !== 'EMOJI-PICKER') {
+      console.log('EmojiPicker: Something wrong. Trigger element is not detected :( But emoji was clicked:', emoji, name);
+      return;
+    }
+    trigger.insertEmoji(emoji);
+  });
+
+  if (nightModeOption) {
+    nightModeOption.addEventListener("optionChanged", () => {
```

Comment:
Please note this way of reacting to the option change. I like it much better than having it in the theme_controller

Reply:
Yes, that's neat.